pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/cuda_kernels_measures.cu

```diff
@@ -0,0 +1,172 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file cuda_kernels_measures.cu
+ */
+#include <cuComplex.h>
+
+#include "cuError.hpp"
+#include "cuda_helpers.hpp"
+
+namespace Pennylane::LightningTensor::TNCuda::Measures {
+/**
+ * @brief Explicitly get the probability of given state tensor data on GPU
+ * device.
+ *
+ * @param state Complex data pointer of state tensor on device.
+ * @param probs The probability result on device.
+ * @param data_size The length of state tensor on device.
+ * @param thread_per_block Number of threads set per block.
+ * @param stream_id Stream id of CUDA calls
+ */
+void getProbs_CUDA(cuComplex *state, float *probs, const int data_size,
+                   const std::size_t thread_per_block, cudaStream_t stream_id);
```

Comment:
Do we need to forward declare those, or can we just use the definitions below?

Reply:
Let's put declare and definition together.